core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -393,6 +393,12 @@ public static SqlOperandTypeChecker variadic(
           // Second operand optional (operand index 0, 1)
           number -> number == 1);
 
+  public static final SqlOperandTypeChecker NUMERIC_INT32 =
+      sequence("ROUND(<NUMERIC>, <INTEGER>)",
+          family(SqlTypeFamily.NUMERIC),
+          // Only 32-bit integer allowed for the second argument
+          new TypeNameChecker(SqlTypeName.INTEGER));
```

Comment:
It's okay for me to use `TypeNameChecker`, but if Calcite implicitly casts other integer types to 32-bit integer, it would be better for this case.

Reply:
These rules explain how ROUND is typechecked.
These rules do not say anything about implicit casts, only about legal arguments accepted by this function.
I believe that type coercion is the one that inserts the implicit casts if they are necessary.